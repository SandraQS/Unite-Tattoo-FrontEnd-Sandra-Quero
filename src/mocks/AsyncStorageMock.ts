import AsyncStorage from "@react-native-async-storage/async-storage";

export const AsyncStorageMock = ()=>{

const mockReturnValues: any = {
  token: JSON.stringify("token"),
};

return jest.mock("@react-native-async-storage/async-storage", () => ({
  setItem: jest.fn(() => {
    return new Promise((resolve) => {
      resolve(null);
    });
  }),
  multiSet: jest.fn(() => {
    return new Promise((resolve) => {
      resolve(null);
    });
  }),
  getItem: jest.fn((key) => {
    return new Promise((resolve) => {
      if (mockReturnValues[key]) {
        resolve(mockReturnValues[key]);
      } else {
        resolve(null);
      }
    });
  }),
  removeItem: jest.fn(() => {
    return new Promise((resolve) => {
      resolve(null);
    });
  }),
}));
}

