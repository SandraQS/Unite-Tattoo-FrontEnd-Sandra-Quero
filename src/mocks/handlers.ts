import { rest } from "msw";
import { REACT_APP_URL_API_UNITETATTOO } from "@env";

const works = [
  {
    tittle: "Pajaritos",
    tattooArtist: "Javi Wolf",
    description: "descripci칩n",
    tattooStyles: "Acuarela",
    likes: 0,
    image:
      "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638124122682-.png",
    collectionWork: "61a09756fe38230f0ba12827",
    id: "61a3b755afd97834ed4aca6e",
  },
  {
    tittle: "Lobo",
    tattooArtist: "Javi Wolff",
    description: "Loremm",
    tattooStyles: "Acuarela",
    likes: 0,
    image:
      "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638119881947-.png",
    collectionWork: "61a37f78b0e47a2b50e4635b",
    id: "61a3b9ca4cc19250ca826626",
  },
];

const user = {
  tattooArtistUser: {
    personalDataTattoArtist: {
      name: "Sandra",
      surname1: "Quero",
      surname2: "S치nchez",
    },
    userDataTattoArtist: {
      userName: "ShivaShana",
      password: "$2b$10$obzQkOXnHi.IyJZtst8FC.4CNK1XlliFmSqJ9VyHPmgBNmPu.fbO2",
      email: "email@gmail.com",
    },
    professionalDataTattooArtist: {
      studioName: "Shiva",
      professionalName: "ShivaShana",
      phone: 666666666,
      contactEmail: "email@gmail.com",
      openingHours: "de 9.00 a 18.00h",
      direction: "C/hola, n췈13",
      tattooStyles: [],
      colaboration: "true",
    },
    collections: [
      {
        tattooStyles: "Realista",
        image:
          "https://storage.googleapis.com/unite-tattoo.appspot.com/Realista-1638921273323-.png",
        works: [
          {
            tittle: "Lobos - Espalda",
            tattooArtist: "Bruno",
            description:
              "Despue패s de un largo tiempo pude terminar esta espalda, una ma패s para la coleccio패n 游때游녧游낖游녧游낖\n俱데fter quite a long time I was able to finish this back piece, another one!!\n\n郊쀮잺TATTOOING IN @circatattoobcn BARCELONA, SPAIN.\n\n@radiantcolorsink\n@bepantholtattoo\n@tsunami_tattoo_needles",
            tattooStyles: "Realista",
            likes: 0,
            image:
              "https://storage.googleapis.com/unite-tattoo.appspot.com/Realista-1638924401230-.png",
            collectionWork: "61aff43a77a75540f78b8449",
            id: "61b0007377a75540f78b8cc3",
          },
        ],
        id: "61aff43a77a75540f78b8449",
      },
      {
        tattooStyles: "Acuarela",
        image:
          "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638923893558-.png",
        works: [],
        id: "61affe7677a75540f78b8569",
      },
    ],
    appointmentSchedule: [],
    imageAmbient: "nada",
    profileImage:
      "https://cdn.pixabay.com/photo/2016/04/01/10/11/avatar-1299805_1280.png",
    id: "61a37f0adaf203887dd7642c",
  },
};

export const handlers = [
  rest.get(
    `${REACT_APP_URL_API_UNITETATTOO}/tattooArtist/collections`,
    async (req, res, ctx) => {
      const response = res(ctx.status(200), ctx.json(user));
      return response;
    }
  ),

  rest.get(
    `${REACT_APP_URL_API_UNITETATTOO}/client/home`,
    async (req, res, ctx) => {
      const response = res(ctx.status(200), ctx.json(works));
      return response;
    }
  ),

  rest.post(
    `${REACT_APP_URL_API_UNITETATTOO}/tattooArtist/collection/create`,
    async (req, res, ctx) => {
      const response = res(
        ctx.status(200),
        ctx.json([
          {
            tattooStyles: "Realista",
            image:
              "https://storage.googleapis.com/unite-tattoo.appspot.com/Realista-1638921273323-.png",
            works: [
              {
                tittle: "Lobos - Espalda",
                tattooArtist: "Bruno",
                description:
                  "Despue패s de un largo tiempo pude terminar esta espalda, una ma패s para la coleccio패n 游때游녧游낖游녧游낖\n俱데fter quite a long time I was able to finish this back piece, another one!!\n\n郊쀮잺TATTOOING IN @circatattoobcn BARCELONA, SPAIN.\n\n@radiantcolorsink\n@bepantholtattoo\n@tsunami_tattoo_needles",
                tattooStyles: "Realista",
                likes: 0,
                image:
                  "https://storage.googleapis.com/unite-tattoo.appspot.com/Realista-1638924401230-.png",
                collectionWork: "61aff43a77a75540f78b8449",
                id: "61b0007377a75540f78b8cc3",
              },
            ],
            id: "61aff43a77a75540f78b8449",
          },
          {
            tattooStyles: "Acuarela",
            image:
              "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638923893558-.png",
            works: [],
            id: "61affe7677a75540f78b8569",
          },
        ])
      );
      return response;
    }
  ),
];
