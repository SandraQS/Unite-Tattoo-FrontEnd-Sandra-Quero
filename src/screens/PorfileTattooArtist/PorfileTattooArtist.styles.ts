import { StyleSheet } from "react-native";
import { colors } from "../../styles/uniteTatto.styles";

export default StyleSheet.create({
  headerPorfile: {
    position: "relative",
  },
  imageAmbient: {
    height: 250,
    opacity: 0.5,
    width: "100%",
  },

  imageBackground: {
    height: 150,
    position: "relative",
    top: -90,
    width: 150,
  },

  imagePorfile: {
    height: 120,
    left: -20,
    position: "absolute",
    top: -20,
    transform: [{ rotateY: "0deg" }, { rotateZ: "-45deg" }],
    width: 120,
  },

  imagePorfileContainer: {
    bottom: 19,
    height: 82,
    left: 34,
    overflow: "hidden",
    position: "absolute",
    transform: [{ rotateY: "25deg" }, { rotateZ: "45deg" }],
    width: 82,
  },

  nameContainer: {
    bottom: 50,
    left: 150,
    position: "absolute",
  },

  nameTattooArtist: {
    fontSize: 35,
    fontWeight: "bold",
    textTransform: "uppercase",
  },
  secondName: {
    fontWeight: "300",
  },
});
