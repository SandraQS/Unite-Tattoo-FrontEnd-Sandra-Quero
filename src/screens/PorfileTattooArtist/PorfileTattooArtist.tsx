import { useNavigation } from "@react-navigation/core";
import React, { useEffect } from "react";
import {
  Image,
  ImageBackground,
  SafeAreaView,
  ScrollView,
  Text,
  View,
} from "react-native";
import NavHeader from "../../components/NavHeader/NavHeader";
import Title from "../../components/Title/Title";
import { useTattooArtist } from "../../hooks/useTattooArtist";
import generalStyles from "../../styles/general.styles";
import styles from "./PorfileTattooArtist.styles";

import { PorfileTattooArtistScreenNavigationProp } from "../../types/navigation.types";

export const PorfileTattooArtist = () => {
  const { tattooArtist, tattooArtistPorfile } = useTattooArtist();
  const navigation = useNavigation<PorfileTattooArtistScreenNavigationProp>();

  useEffect(() => {
    tattooArtistPorfile();
  }, []);

  return (
    <SafeAreaView style={generalStyles.screenLightBrown}>
      <View style={generalStyles.navHeader}>
        <NavHeader client={false} goBack={() => navigation.pop()} />
      </View>
      <ScrollView>
        <View style={styles.headerPorfile}>
          <View>
            <Image
              source={{ uri: tattooArtist.imageAmbient }}
              style={styles.imageAmbient}
            />
          </View>

          <View>
            <ImageBackground
              source={require("../../assets/marco-rombo.png")}
              resizeMode="contain"
              style={styles.imageBackground}
            >
              <View style={styles.imagePorfileContainer}>
                <Image
                  source={{ uri: tattooArtist.profileImage }}
                  style={styles.imagePorfile}
                  resizeMode="cover"
                />
              </View>
            </ImageBackground>
          </View>

          <View style={styles.nameContainer}>
            <Text style={styles.nameTattooArtist}>
              {tattooArtist.personalDataTattoArtist.name}
            </Text>
            <Text style={[styles.nameTattooArtist, styles.secondName]}>
              {tattooArtist.personalDataTattoArtist.surname1}
            </Text>
          </View>
        </View>

        <View style={generalStyles.mainContainerGeneral}>
          <View>
            <Title text="HORARIO" />
            <Text>{`${tattooArtist.professionalDataTattooArtist.openingHours}`}</Text>
          </View>

          <View>
            <Title text="DIRECCIÓN" />
            <Text>{`${tattooArtist.professionalDataTattooArtist.direction}`}</Text>
          </View>

          <View>
            <Title text="ABIERTO A COLABORACIONES" />
            <Text>
              {tattooArtist.professionalDataTattooArtist.colaboration
                ? "Sí"
                : "No"}
            </Text>
          </View>

          <View>
            <Title text="EMAIL" />
            <Text>{`${tattooArtist.professionalDataTattooArtist.contactEmail}`}</Text>
          </View>

          <View>
            <Title text="TELÉFONO" />
            <Text>{`${tattooArtist.professionalDataTattooArtist.phone}`}</Text>
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};
