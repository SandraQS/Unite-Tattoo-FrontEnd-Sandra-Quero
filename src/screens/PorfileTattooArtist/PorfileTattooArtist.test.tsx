import React from "react";
import { server } from "../../mocks/server";
import { render, waitFor } from "@testing-library/react-native";
import configureStore from "../../redux/store";
import { Provider } from "react-redux";
import { NavigationContainer } from "@react-navigation/native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { PorfileTattooArtist } from "./PorfileTattooArtist";

jest.mock("@react-native-async-storage/async-storage", () => ({
  setItem: jest.fn(),
  getItem: jest.fn(),
  removeItem: jest.fn(),
}));

beforeAll(() => {
  server.listen({
    onUnhandledRequest: "warn",
  });
});

afterEach(() => {
  server.resetHandlers();
});

afterAll(() => {
  server.close();
});
describe("Give PorfileTattooArtist screen", () => {
  describe("When it's rendered", () => {
    test("Then it should show match with the last snapsot", async () => {
      const store = configureStore();
      AsyncStorage.getItem = jest.fn().mockResolvedValue({ token: "algo" });

      const screen = render(
        <Provider store={store}>
          <NavigationContainer>
            <PorfileTattooArtist />
          </NavigationContainer>
        </Provider>
      );

      await waitFor(() => {
        expect(screen).toMatchSnapshot();
      });
    });
  });
});
