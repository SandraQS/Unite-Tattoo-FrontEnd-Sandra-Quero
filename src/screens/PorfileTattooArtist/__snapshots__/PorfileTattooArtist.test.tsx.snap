// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Give PorfileTattooArtist screen When it's rendered Then it should show match with the last snapsot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#DFD8D2",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#7C6B53",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 20,
          "paddingTop": 50,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            onError={[Function]}
            source={
              Object {
                "testUri": "../../../src/assets/utoo-logo.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 0,
                  "width": 100,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "color": "#7C6B53",
                  "fontSize": 15,
                  "paddingEnd": 10,
                }
              }
            >
              Salir
            </Text>
          </View>
          <View
            style={Object {}}
          >
            <Image
              onError={[Function]}
              source={
                Object {
                  "testUri": "../../../src/assets/icono-user.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 0,
                    "width": 30,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <View>
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "height": 250,
                "opacity": 0.5,
                "width": "100%",
              }
            }
          />
        </View>
        <View>
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": 150,
                "position": "relative",
                "top": -90,
                "width": 150,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../src/assets/marco-rombo.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 150,
                    "width": 150,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Object {
                  "bottom": 19,
                  "height": 82,
                  "left": 34,
                  "overflow": "hidden",
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotateY": "25deg",
                    },
                    Object {
                      "rotateZ": "45deg",
                    },
                  ],
                  "width": 82,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "height": 120,
                    "left": -20,
                    "position": "absolute",
                    "top": -20,
                    "transform": Array [
                      Object {
                        "rotateY": "0deg",
                      },
                      Object {
                        "rotateZ": "-45deg",
                      },
                    ],
                    "width": 120,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
